File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/apache/ApacheResponse.java
package ru.kmorozov.gbd.core.logic.connectors.apache;

import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
import org.apache.hc.core5.http.io.entity.EntityUtils;
import ru.kmorozov.gbd.core.logic.connectors.Response;

import java.io.IOException;
import java.io.InputStream;

/**
 * Created by km on 20.05.2016.
 */
public class ApacheResponse implements Response {

    private final CloseableHttpResponse response;

    ApacheResponse(final CloseableHttpResponse response) {
        this.response = response;
    }

    @Override
    public InputStream getContent() throws IOException {
        return null == response ? null : response.getEntity().getContent();
    }

    @Override
    public String getImageFormat() {
        final String contentType = response.getEntity().getContentType();

        return contentType.startsWith("image/") ? contentType.split("/")[1] : null;
    }

    @Override
    public void close() {
        if (null != response) EntityUtils.consumeQuietly(response.getEntity());
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/connectors/apache/ApacheResponse.kt
package ru.kmorozov.gbd.core.logic.connectors.apache

import org.apache.http.client.methods.CloseableHttpResponse
import org.apache.http.util.EntityUtils
import ru.kmorozov.gbd.core.logic.connectors.Response

import java.io.IOException
import java.io.InputStream

/**
 * Created by km on 20.05.2016.
 */
class ApacheResponse internal constructor(private val response: CloseableHttpResponse) : Response {
    override val statusCode: Int
        get() = response.statusLine.statusCode

    override val content: InputStream
        @Throws(IOException::class)
        get() = response.entity.content

    override val headers: String
        get() = response.allHeaders.toString()

    override val imageFormat: String
        get() {
            val contentType = response.entity.contentType.value

            return if (contentType.startsWith("image/")) contentType.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[1] else "unknown"
        }

    override fun close() {
        EntityUtils.consumeQuietly(response.entity)
    }
}
